esmvalcore/preprocessor/_area.py

```diff
@@ -913,3 +913,72 @@ def _mask_cube(cube: Cube, masks: dict[str, np.ndarray]) -> Cube:
         for ancillary_variable in cube.ancillary_variables():
             add_ancillary_variable(result, ancillary_variable)
     return result
+
+
+@register_supplementaries(
+    variables=['areacella', 'areacello'],
+    required='prefer_at_least_one',
+)
+def normalize_with_area_statistics(
+    cube: Cube,
+    operation: str = 'subtract',
+    area_statistics_kwargs: Optional[dict] = None,
+) -> Cube:
+    """Normalize data with output from ``area_statistics``.
+
+    For example, this can be used to subtract the global mean from a field or
+    divide by the global standard deviation.
+
+    Parameters
+    ----------
+    cube:
+        Input cube that is normalized. The input cube should have a
+        :class:`iris.coords.CellMeasure` named ``'cell_area'``, unless it has
+        regular 1D latitude and longitude coordinates so the cell areas can be
+        computed using :func:`iris.analysis.cartography.area_weights`.
+    operation:
+        Normalization operation, can be `subtract` or `divide`.
+    area_statistics_kwargs:
+        Keyword arguments for :func:`esmvalcore.preprocessor.area_statistics`.
+        The output from this operation is used to normalize the input cube. If
+        ``None``, use ``{'operator': 'mean'}``.
+
+    Returns
+    -------
+    iris.cube.Cube
+        Normalized cube.
+
+    Raises
+    ------
+    ValueError
+        `operation` is not one of `subtract`, `divide`.
+    iris.exceptions.CoordinateMultiDimError
+        Cube has irregular or unstructured grid but supplementary variable
+        `cell_area` is not available.
+
+    """
+    # Create cube that is used to normalize the input cube
+    if area_statistics_kwargs is None:
+        area_statistics_kwargs = {'operator': 'mean'}
+    area_stats_cube = area_statistics(cube, **area_statistics_kwargs)
+
+    # Normalize cube
+    # Note: the area statistics cube is always broadcastable to the input cube
+    # due to iris' broadcasting rules enabled by the named dimensions
+    metadata = cube.metadata
+    if operation == 'subtract':
+        cube = cube - area_stats_cube
+    elif operation == 'divide':
+        cube = cube / area_stats_cube
```

Comment:
do you want to catch a zero-division warning here and perform a check on the `area_stats_cube` if that pops up?

Reply:
Good point. I just tested this; with the current version of the code this already raises an `RuntimeWarning: invalid value encountered in divide` warning and gives `nan` values in the output. I can add a nicer message here. Would it also make sense to mask the cube instead of the `nan`s? I am not sure, maybe it also makes sense to keep the `nan`s to properly distinguish between the real mask and the zero-divisions.

Reply:
I found a weird problem with iris: https://github.com/SciTools/iris/issues/5523. Sometimes the output of zero divisions is masked, sometimes not. Thus, I chose to mask them consistently here.

Reply:
so in light of that issue, let's wait for the linked iris issue to be fixed before we do anything ourselves - in the lower stacks - it's fine to fix here (I reckon) :beers: